ctx/context.go

```diff
@@ -6,110 +6,158 @@ import (
 	"path"
 	"sync"
 
-	appctx "github.com/nixys/nxs-go-appctx/v2"
+	"github.com/hashicorp/go-multierror"
+	appctx "github.com/nixys/nxs-go-appctx/v3"
+	"github.com/sirupsen/logrus"
 
-	"nxs-backup/interfaces"
-	"nxs-backup/modules/logger"
+	"github.com/nixys/nxs-backup/interfaces"
+	"github.com/nixys/nxs-backup/modules/cmd_handler"
+	"github.com/nixys/nxs-backup/modules/logger"
 )
 
 // Ctx defines application custom context
 type Ctx struct {
-	ConfigPath   string
-	CmdParams    interface{}
-	Storages     interfaces.Storages
-	Jobs         interfaces.Jobs
-	FilesJobs    interfaces.Jobs
-	DBsJobs      interfaces.Jobs
-	ExternalJobs interfaces.Jobs
-	LogCh        chan logger.LogRecord
-	Notifiers    []interfaces.Notifier
-	WG           *sync.WaitGroup
-
-	Cfg confOpts
+	Cmd       interfaces.Handler
+	Log       *logrus.Logger
+	Done      chan error
+	EventCh   chan logger.LogRecord
+	EventsWG  *sync.WaitGroup
+	Notifiers []interfaces.Notifier
 }
 
-// Init initiates application custom context
-func (c *Ctx) Init(opts appctx.CustomContextFuncOpts) (appctx.CfgData, error) {
+func AppCtxInit() (any, error) {
```

Comment:
Try to use `switch` statement to make this function more structural

Reply:
done